### PR TITLE
Clean up command handling in hdp.c/h

### PR DESCRIPTION
**Pull request by derobins**
* Move list of commands inside hdp.c
* Remove HDP_MASTER scheme
* Tie command string & enum value together in a struct to keep in sync
* Minor tidying

### CODE REVIEW

File: mfhdf/dumper/hdp.h [hunk 1]
```diff
@@ -283,18 +278,6 @@ extern
...
-/* Preserve the correct/corresponding ordering */
-typedef enum { HELP, LIST, DUMPSDS, DUMPRIG, DUMPVG, DUMPVD, DUMPGR, NONE } command_t;
-#ifndef HDP_MASTER
-extern
-#endif /* !HDP_MASTER */
-    const char *commands[]
-#ifdef HDP_MASTER
-    = {"help", "list", "dumpsds", "dumprig", "dumpvg", "dumpvd", "dumpgr"}
-#endif /* HDP_MASTER */
-;
```
Comment (derobins) [inline mfhdf/dumper/hdp.h:296]: Moved into hdp.c since it's not used anywhere else

File: mfhdf/dumper/hdp.c [hunk 2]
```diff
@@ -11,11 +11,28 @@
...
+/* indicates Vsets have been initialized for the current file */
+intn vinit_done = FALSE;
+
+/* hdp commands (stored as (value, name) pairs to keep them in sync) */
+typedef enum { HELP, LIST, DUMPSDS, DUMPRIG, DUMPVG, DUMPVD, DUMPGR, BAD_COMMAND } command_value_t;
+
+typedef struct command_t {
+    const command_value_t value;
+    const char           *name;
+} command_t;
```
Comment (derobins) [inline mfhdf/dumper/hdp.c:29]: Tying the enum value and string together into a struct ensures they will never be out of sync

File: mfhdf/dumper/hdp.c [hunk 3]
```diff
@@ -87,58 +110,56 @@ init_dump_opts(dump_info_t *dump_opts)
...
+        if (HDstrcmp(argv[curr_arg], commands[j].name) == 0) {
+            cmd = commands[j].value;
             break;
-    } /* end for */
+        }
+    }
+    if (cmd == BAD_COMMAND) {
+        printf("Invalid command: %s\n", argv[curr_arg]);
+        exit(EXIT_FAILURE);
+    }
```
Comment (derobins) [inline mfhdf/dumper/hdp.c:150]: Before we printed the value of cmd (in code below), which would just be the sentinel NONE value, which is unhelpful.

File: mfhdf/dumper/hdp.c [hunk 4]
```diff
@@ -148,36 +169,37 @@ main(int argc, char *argv[])
...
 
+        /* Should not get here - invalid commands are handled earlier */
+        case BAD_COMMAND:
         default:
-            printf("Invalid command!, cmd=%d\n", (int)cmd);
-            exit(1);
+            printf("Invalid command!\n");
+            exit(EXIT_FAILURE);
             break;
-    } /* end switch */
```
Comment (derobins) [inline mfhdf/dumper/hdp.c:178]: Printing the enum value was unhelpful. Printing the string is better, which we do above.

**Comment by derobins** [comment]
Test failures are due to GitHub flakiness and aren't real

**Review by bmribler — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**